src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -1315,13 +1316,26 @@ def migrate_database(self) -> None:
                         backup_location,
                     ) = self.backup_database(overwrite=True)
                 except Exception as e:
-                    raise RuntimeError(
-                        f"Failed to backup the database: {str(e)}. "
-                        "Please check the logs for more details."
-                        "If you would like to disable the database backup "
-                        "functionality, set the `backup_strategy` attribute "
-                        "of the store configuration to `disabled`."
-                    ) from e
+                    # The database backup feature was not entirely functional
+                    # in ZenML 0.56.3 and earlier, due to inconsistencies in the
+                    # database schema. If the database is at version 0.56.3
+                    # or earlier and if the backup fails, we only log the
+                    # exception and leave the upgrade process to proceed.
+                    if version.parse(current_revisions[0]) <= version.parse(
+                        "0.56.3"
```

Comment:
This does full semantic comparision?

Reply:
Yes, but you're right about one thing: this will break if not running on an actual official release. Let me fix that.

Reply:
Done